_template_fastapi_/provider/client.py

```diff
@@ -0,0 +1,40 @@
+"""
+Client class to retrieve data from a custom data source
+"""
+from datetime import datetime
+from datamodels import DataItem
+
+class CustomClient:
+    def __init__(self):
+        """
+        Initialize the client
+        """
+        pass
+
+    def search(self, query: str):
+        """
+        Logic to retrieve data based on query
+
+        Args:
+            query (str): Query string
+
+        Returns:
+            List[dict]: List of data items
+        """
+        data = [
+            {
+                "id": 1,
+                "url": "https://example.com/article1",
+                "title": "The Future of Technology",
+                "text": "An in-depth analysis of emerging technology trends...",
+                "timestamp": "2024-01-30 15:30:00"
+            },
+            {
+                "id": 2,
+                "url": "https://example.com/article2",
+                "title": "Exploring the Natural World",
+                "text": "A journey through the wonders of our planet...",
+                "timestamp": "2024-01-31 10:00:00"
+            }
+        ]
+        return data
```

Comment:
Could you add a `get_client()` method below here that would act as a helper method to help instantiate `CustomClient`? 

This is where environment variables needed for the client can be checked, andf or connectors that don't offer OAuth, we also generally have a `global client` var to avoid instantiating the client multiple times. Quick example:

```
def get_client():
    global client
    if client is not None:
        return client

    assert (myvar := os.getenv("CONNECTOR_VAR")), "CONNECTOR_VAR must be set"
    client = CustomClient(myvar)

    return client

Reply:
@tianjing-li thanks for the heads-up! will add get_client()